examples/bert/run_glue_finetuning.py

```diff
@@ -200,28 +198,60 @@ def preprocess_data(inputs, labels):
 
     # Read and preprocess GLUE task data.
     train_ds, test_ds, validation_ds = load_data(FLAGS.task_name)
-    train_ds = train_ds.batch(FLAGS.batch_size).map(
+
+    # batch_size is taken as 32.
+    train_ds = train_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
-    validation_ds = validation_ds.batch(FLAGS.batch_size).map(
+    validation_ds = validation_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
-    test_ds = test_ds.batch(FLAGS.batch_size).map(
+    test_ds = test_ds.batch(batch_size).map(
         preprocess_data, num_parallel_calls=tf.data.AUTOTUNE
     )
 
-    finetuning_model = BertClassificationFinetuner(
-        bert_model=model,
-        hidden_size=bert_config["hidden_size"],
-        num_classes=3 if FLAGS.task_name in ("mnli", "ax") else 2,
+    class BertHyperModel(keras_tuner.HyperModel):
+        """Creates a hypermodel to help with the search space for finetuning."""
+
+        def build(self, hp):
+            finetuning_model = BertClassificationFinetuner(
+                bert_model=model,
+                hidden_size=bert_config["hidden_size"],
+                num_classes=3 if FLAGS.task_name in ("mnli", "ax") else 2,
+            )
+            finetuning_model.compile(
+                optimizer=keras.optimizers.Adam(
+                    learning_rate=hp.Choice("lr", [5e-5, 4e-5, 3e-5, 2e-5])
+                ),
+                loss="sparse_categorical_crossentropy",
+                metrics=["accuracy"],
+            )
+            return finetuning_model
+
+    # Create a hypermodel object for a RandomSearch.
+    hypermodel = BertHyperModel()
+
+    # Initialize the random search over the 4 learning rate parameters, for 4
+    # trials and 3 epochs for each trial.
+    tuner = keras_tuner.RandomSearch(
+        hypermodel=hypermodel,
+        objective=keras_tuner.Objective("val_loss", direction="min"),
+        max_trials=4,
+        overwrite=True,
+        directory="tuning_hp_bert",
+        project_name="glue_finetuning_hp",
     )
-    finetuning_model.compile(
-        optimizer=keras.optimizers.Adam(learning_rate=FLAGS.learning_rate),
-        loss="sparse_categorical_crossentropy",
-        metrics=["accuracy"],
-    )
-    finetuning_model.fit(
-        train_ds, epochs=FLAGS.epochs, validation_data=validation_ds
+
+    tuner.search(train_ds, epochs=epochs, validation_data=validation_ds)
+
+    # Extract the best hyperparameters after the search.
+    best_hp = tuner.get_best_hyperparameters()[0]
+    finetuning_model = tuner.get_best_models()[0]
+
+    finetuning_model.fit(train_ds, epochs=epochs, validation_data=validation_ds)
+
+    print(
```

Comment:
@mattdangerw Do we want to encourage using `logging` in our examples?

Reply:
I would say simple print is more accessible until we find a concrete reason to move to logging. We should expect this to be run locally and on GCP. Is there an issue with a simply printing in either case?